server/src/pages/shot/view.js

```diff
@@ -393,9 +395,11 @@ class Body extends React.Component {
           <div className="shot-main-actions">
             <div className="shot-info">
               <EditableTitle title={shot.title} isOwner={this.props.isOwner} />
-              <div className="shot-subtitle"> { favicon }
-                { linkTextShort ? <a className="subtitle-link" rel="noopener noreferrer" href={ shot.url } target="_blank" onClick={ this.onClickOrigUrl.bind(this, "navbar") }>{ linkTextShort }</a> : null }
-                <span className="time-diff">{ timeDiff }</span> { expiresDiff }
+              <div className="shot-subtitle">
+                { this.state.isChangingExpire ? null : favicon }
+                { linkTextShort && !this.state.isChangingExpire ? <a className="subtitle-link" rel="noopener noreferrer" href={ shot.url } target="_blank" onClick={ this.onClickOrigUrl.bind(this, "navbar") }>{ linkTextShort }</a> : null }
+                { this.state.isChangingExpire ? null : <span className="time-diff">{ timeDiff }</span> }
```

Comment:
What if, instead of repeating three `this.state.isChangingExpire` checks, we wrap these items in a div that can have the `hidden` attribute toggled if `this.state.isChangingExpire`?

Fine to land as-is, though.